tools/depends/configure.ac

```diff
@@ -418,7 +418,7 @@ case $host in
     platform_cxxflags+=" -arch $use_cpu -m$platform_min_version -stdlib=libc++"
     platform_includes="-isysroot $use_sdk_path"
     deps_dir="${sdk_name}_${use_cpu}-target-${build_type}"
-    AC_CHECK_LIB([z], [main], has_zlib=1, AC_MSG_WARN("No zlib support in toolchain. Will build libz."); has_zlib=0)
+    AC_SEARCH_LIBS([deflate],z, link_zlib=$ac_cv_search_zlib_main, link_zlib=-lz; AC_MSG_WARN("No zlib support in toolchain. Will build zlib."); has_zlib=1)
```

Comment:
Can you explain this?

Reply:
absolutely may be a better way to do this. Im just not sure.

Context is, we need to add -lz to linker flags (in particular for this use case CMAKE_EXE_LINKER_FLAGS for use by mariadb-connector-c). 
AC_SEARCH_LIBS appears to append to $LIBS as per https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Libraries.html

So essentially, checking for deflate function in the z library, 3rd option is on found, as its pre-depends, its actually not found for ios/tvos, so this argument is really null anyway. last argument is on failure to find. I think ive got the has_zlib=1 wrong, i think it should be 0, i'll have to double check that. Either way, this adds -lz to $LIBS, which is then added to platform_ldflags at https://github.com/xbmc/xbmc/blob/c4405521ac0f1f8efacda83d4b0b2cf21bc7a7cc/tools/depends/configure.ac#L635

Realistically, i dont know the history behind the usage of ac_check_lib (again, it will always do the fail case of 4th argument anyway for ios/tvos), so we may just be able to add -lz explicitly to platform_ldflags and call it a day. Maybe need to set has_zlib=0, not totally sure where that gets used past configure

Reply:
Ok, that explains why we don’t need this in a binary addon as we explicitly build the dependencies.

Reply:
yeah, thats the funny thing, we do in core as well, haha. i think the usage is totally redundant

Reply:
Is their no package config file for Mariadb connector c?

Reply:
there is, just doesnt seem to link whats needed though.

Reply:
Why not patch the PC file instead?

Reply:
And then the next dependency that needs it as well? just easier to set it globally imo.

Reply:
my understanding was a little flawed from my earlier comment.
I've fixed it up to be more accurate.

Essentially on darwin platforms, we wont actually build/use target zlib, as system zlib (10.14 with xcode 10.3 is 1.2.11 which is what we currently use in target depends).
The search finds the toolchain zlib, so we still add the linker flag using ac_search_libs. We could patch the pc file, but i think this is more robust long term, without carrying/maintaining a patch.